module/documents/actor/trait.mjs

```diff
@@ -232,24 +419,109 @@ export function keyLabel(trait, key) {
+ *
+ * @example
+ * // Returns "Thieves' Tools or any artisan tool"
+ * choiceLabel({ count: 1, pool: new Set(["tool:thief", "tool:art:*"]) }, { only: true });
+ *
+ * @example
+ * // Returns
+ * choiceLabel({ count: 2, pool: new Set(["tool:thief", "skills:*"]) });
+ *
  */
-export function choiceLabel(trait, choice) {
-  // Select from any trait values
-  if ( !choice.pool ) {
-    return game.i18n.format("DND5E.TraitConfigChooseAny", {
-      count: choice.count,
-      type: traitLabel(trait, choice.count).toLowerCase()
+export function choiceLabel(choice, { only=false, final=false }={}) {
+  if ( !choice.pool.size ) return "";
+
+  // Single entry in pool (e.g. "any three skill proficiencies" or "three other skill proficiencies")
+  if ( choice.pool.size === 1 ) {
+    return keyLabel(choice.pool.first(), {
+      count: (choice.count > 1 || !only) ? choice.count : null, final: final && !only
     });
   }
 
-  // Select from a list of options
-  const choices = choice.pool.map(key => keyLabel(trait, key));
-  const listFormatter = new Intl.ListFormat(game.i18n.lang, { type: "disjunction" });
+  const listFormatter = game.i18n.getListFormatter({ type: "disjunction" });
+
+  // Singular count (e.g. "any skill", "Thieves Tools or any skill", or "Thieves' Tools or any artisan tool")
+  if ( (choice.count === 1) && only ) {
+    return listFormatter.format(choice.pool.map(key => keyLabel(key)));
+  }
+
+  // Select from a list of options (e.g. "Choose 2 from thieves tools or any skill proficiency")
+  const choices = choice.pool.map(key => keyLabel(key));
   return game.i18n.format("DND5E.TraitConfigChooseList", {
     count: choice.count,
     list: listFormatter.format(choices)
   });
 }
+
+/* -------------------------------------------- */
+
+/**
+ * Create a human readable description of trait grants & choices.
+ * @param {Set<string>} grants                                        Guaranteed trait grants.
+ * @param {TraitChoice[]} [choices=[]]                                Trait choices.
+ * @param {object} [options={}]
+ * @param {"inclusive"|"exclusive"} [options.choiceMode="inclusive"]  Choice mode.
+ * @returns {string}
+ *
+ * @example
+ * // Returns "Acrobatics and Athletics"
+ * localizedList(new Set(["skills:acr", "skills:ath"]));
+ *
+ * @example
+ * // Returns "Acrobatics and one other skill proficiency"
+ * localizedList(new Set(["skills:acr"]), [{ count: 1, pool: new Set(["skills:*"])}]);
+ *
+ * @example
+ * // Returns "Choose any skill proficiency"
+ * localizedList(new Set(), [{ count: 1, pool: new Set(["skills:*"])}]);
+ *
+ * @example
+ * // Returns "Choose any 2 languages or any 1 skill proficiency"
+ * localizedList(new Set(), [
+ *   {count: 2, pool: new Set(["languages:*"])}, { count: 1, pool: new Set(["skills:*"])}
+ * ], {choiceMode: "exclusive"});
+ */
+export function localizedList(grants, choices=[], { choiceMode="inclusive" }={}) {
```

Comment:
Since these seem all kind of optional from the examples, maybe it should just be one big options object:

```suggestion
export function localizedList({ grants=new Set(), choices=[], choiceMode="inclusive" }={}) {
```